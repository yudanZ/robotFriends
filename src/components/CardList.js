import React from 'react';
import Card from './Card';
import './CardList.css';


const CardList = ({ robots }) => {
    return (
        <div className="robotsContainer" >
            {
                robots.map( (el, i) => {
                    return (
                        <Card 
                            key={i} 
                            id={robots[i].id} 
                            name={robots[i].name} 
                            email={robots[i].email} 
                            imgUrl={robots[i].imgUrl}
                        />
                    );
                    
                })   
            }   
        </div>
    );
    
}

export default CardList;