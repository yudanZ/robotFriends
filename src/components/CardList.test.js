import { shallow} from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John Snow',
            username: 'JohnHohn',
            email: 'john@gmail.com',
            imgUrl: 'https://robohash.org/AGP.png?set=set4'
        }
    ]
    expect(shallow(<CardList  robots={mockRobots}/>).debug()).toMatchSnapshot();
})