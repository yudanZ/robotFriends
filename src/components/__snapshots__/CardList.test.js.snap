// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
"<div className=\\"robotsContainer\\">
  <Card id={1} name=\\"John Snow\\" email=\\"john@gmail.com\\" imgUrl=\\"https://robohash.org/AGP.png?set=set4\\" />
</div>"
`;
